### ISSUE TITLE
Mod addonInfo unrecognized

### ISSUE THREAD

**Issue by patri45** [pukmajster/funky#11]
As the title described, some mods don't show any addonInfo (such as name, author, version, description) in the program. I tried to correct addonInfo.txt in .vpk file, but it's still unrecognized. Paste some mods with this problem below:

[VScripts] Library of Utils  https://steamcommunity.com/sharedfiles/filedetails/?id=2821408044
[Improved] Headshot Feedback Effect  https://steamcommunity.com/sharedfiles/filedetails/?id=2582265366
The Particles Manifest  https://steamcommunity.com/sharedfiles/filedetails/?id=1380516683

by the way, if is it possible to display local path and file name of the mod on the right-click menu? 
It's so hard to navigate the problem .vpk file

**Comment by pukmajster** [comment]
This should be resolved by keeping networking enabled in settings, which is enabled by default. Can you check if you have it enabled?
![image](https://github.com/user-attachments/assets/e7bd60bf-b00e-445f-9c9d-c219030d9e83)

.vpk filepaths seems easy to add though.

**Comment by pukmajster** [comment]
Welp, the issue comes from calling `.toString('utf-8')` on the contents of the `addoninfo.txt` file!

Said file in question...

```
// 这个描述文件一定要用 Notepad++ 或其他文本软件去编辑（为了保持编码为 UTF-8无BOM），最好不要用系统自带记事本去编辑，否则编码容易变成ANSI，从而导致中文乱码！
// Notepad++ 请自行下载，其官网有严重国家反动内容，我个人拒绝提供。

"AddonInfo"	//
{  						//

     "addonSteamAppID"							"550"						// 游戏ID（无需修改）
     "addontitle"											"XDR Charger Animation(my asss!) v1.2"			// 标题
     "addonversion"										"1.2"						// 版本号
     "addontagline"										""								// 
     "addonauthor"										"热心网友"						// MOD作者
     "addonauthorSteamID"						""						// MOD作者的SteamID
     "addonContent_Campaign"				"0"							// 是否为地图MOD，1是0否
     "addonContent_Map"							"0"							// 是否为地图MOD
     "addonContent_Skin"							"0"							// 是否为皮肤MOD
     "addonContent_Weapon"					"0"							// 是否为武器MOD
     "addonContent_BossInfected"			"0"							// 是否为地图MOD
     "addonContent_CommonInfected"	"0"							// 是否为小僵尸MOD
     "addonContent_Survivor"					"0"							// 是否为生还者MOD
     "addonContent_Sound"						"0"							// 是否含有声音
     "addonContent_Script"						"0"							// 是否含有脚本
     "addonURL0"										""								// MOD发布链接（可写可不写）
     "addonDescription								"小心不要被牛牛抓到!--by热心网友。"								// MOD描述
	 
}						// ←别动
```

Honestly, at this point I'm ready get rid of local addoninfo.txt parsing and rely fully on the API method, as these files have been nothing but pain to work with.

**Comment by pukmajster** [comment]
Now that I think about it, I think this happens because this feature was **enabled**...

**Comment by pukmajster** [comment]
The reality here is that there's going to be plenty of workshop mods that provide a faulty addoninfo.txt, or none at all, meaning i just won't be able to display anything. In such case, a little notice will be displayed on the mod overview saying that the full metadata is simply not available. 

![image](https://github.com/user-attachments/assets/e6d22468-4703-47e5-8d48-df81f9d23d41)

The new experimental manifest parser prioritizes extracting metadata from the VPK directly, but if that fails, Funky will fetch the mod title from the Steam Workshop API. The new parser is available in release v0.4.0 and must be (for now) enabled through settings.

This is the best solution I've got right now, and I think is acceptable. Closing this issue.
